riscv/src/compiler.rs

```diff
@@ -307,6 +307,23 @@ fn replace_dynamic_label_reference(
     ))
 }
 
+fn filter_matching_and_next<I: Iterator, F>(iter: I, predicate: F) -> impl Iterator<Item = I::Item>
+where
+    F: Fn(&I::Item) -> bool,
+{
+    iter.scan(false, move |filter_next, item| {
+        let mut filter_current = *filter_next;
+        *filter_next = predicate(&item);
+        // if the predicate says this line should be filtered, then
+        // the next one should be filtered as well.
+        if *filter_next {
+            filter_current = true;
+        }
```

Comment:
Good catch, this is even simpler!
We can even push it to `filter_current |= *filter_next`